libethash-cuda/ethash_cuda_miner_kernel.cu

```diff
@@ -38,7 +38,6 @@ ethash_search(
 	if (cuda_swab64(hash) > d_target) return;
 	uint32_t index = atomicInc(const_cast<uint32_t*>(g_output), SEARCH_RESULT_BUFFER_SIZE - 1) + 1;
 	g_output[index] = gid;
-	__threadfence_system();
```

Comment:
That fence was possibly necessary (AFAIK there is no clear documentation on this, i implemented the fence just to be sure) because the g_output array was originally by mistake allocated on host. Since that mistake was fixed and g_output is now properly allocated on device, the fence is no longer necessary.